plotting/plot_handler.py

```diff
@@ -42,20 +44,40 @@ def __init__(self, instrument_name, run_number, server_dir, rb_number=None):
                                              'autoreduce_webapp', 'static',
                                              'graphs')
 
-    def _generate_file_name_regex(self, plot_type):
+    def _generate_file_name_regex(self):
         """
         Regular expression used for looking for plot files.
         This assumes that the file names follow the convention:
         <instrument_abbreviation><run_number>*<.png or other extension>
-        :param plot_type: (str) the type of file to be searched for, e.g. "png"
         """
         try:
             _inst_regex = INSTRUMENT_REGEX_MAP[self.instrument_name]
-            return f'{_inst_regex}{self.run_number}.*.{plot_type}'
+            _file_extension_regex = self._generate_file_extension_regex()
+            return f'{_inst_regex}{self.run_number}.*.{_file_extension_regex}'
         except KeyError:  # Instrument name does not appear in dictionary of known instruments
             LOGGER.info("The instrument name is not recognised")
             return None
 
+    def _generate_file_extension_regex(self):
+        """
+        Generates the file extension part of the file regex. For example if the file extensions were
+        .png, .gif and .jpg: The returned value would be (png|gif|jpg)
+        :return: (str) expression pattern matching the file extensions of the plot handler
+        """
+        _extension_regex = '('
+        for extension in self.file_extensions:
+            _extension_regex += f'{extension}|'
+        return _extension_regex[:-1] + ')'
```

Comment:
Could this not just be replace with something like?  
```suggestion
 return f"({','.join(self.file_extensions).replace(',', '|')})"
```
or:
```suggestion
 file_extensions_as_str = ','.join(self.file_extensions)
 file_extensions_for_regex = file_extensions_as_str.replace(',', '|')
 return f"({file_extensions_for_regex})"
```

Reply:
Much cleaner, after 8 months of writing java I forget python actually has nice string tools 👍